amaranth/build/run.py

```diff
@@ -92,23 +87,59 @@ def execute_local(self, root="build", *, run_script=True, env=None):
-                    # See https://stackoverflow.com/a/30736987 for a detailed explanation of why.
-                    # Running the script manually from a command prompt is unaffected.
-                    subprocess.check_call(["cmd", "/c", f"call {self.script}.bat"],
-                                          env=os.environ if env is None else env)
-                else:
-                    subprocess.check_call(["sh", f"{self.script}.sh"],
-                                          env=os.environ if env is None else env)
-
-            return LocalBuildProducts(os.getcwd())
-
+            return os.getcwd()
         finally:
             os.chdir(cwd)
 
+    def execute_local(self, root="build", *, run_script=True, env=None):
+        """
+        Execute build plan using the local strategy. Files from the build plan are placed in
+        the build root directory ``root``, and, if ``run_script`` is ``True``, the script
+        appropriate for the platform (``{script}.bat`` on Windows, ``{script}.sh`` elsewhere)
+        is executed in the build root. If ``env`` is not ``None``, the environment is replaced
+        with ``env``.
+
+        Returns :class:`LocalBuildProducts`.
+        """
+        build_dir = self.extract(root)
+        if run_script:
+            if sys.platform.startswith("win32"):
+                # Without "call", "cmd /c {}.bat" will return 0.
+                # See https://stackoverflow.com/a/30736987 for a detailed explanation of why.
+                # Running the script manually from a command prompt is unaffected.
+                subprocess.check_call(["cmd", "/c", f"call {self.script}.bat"],
+                                        env=os.environ if env is None else env)
+            else:
+                subprocess.check_call(["sh", f"{self.script}.sh"],
+                                        env=os.environ if env is None else env)
+
+        return LocalBuildProducts(build_dir)
+
+
+    def execute_local_docker(self, root="build", *, container_image, container_workdir="/build",
+                       docker_args=[]):
+        """
+        Execute build plan inside a docker container. Files from the build plan are placed in the 
+        build root directory ``root``, on the local filesystem. if ``run_script`` is ``True``, the
+        build root directory is is bind mounted to ``/build`` in the container and the 
+        script ``{script}.sh`` is executed inside the container.
+
+        ``docker_args`` is a dictionary that holds all arguments to docker
+
+        Returns :class:`LocalBuildProducts`.
+        """
+        build_dir = self.extract(root)
+        subprocess.check_call([
+            "docker", "run", *docker_args,
+            "--rm", # remove the container after running
+            "--mount", f"type=bind,source={os.getcwd()},target={container_workdir}",
+            "--workdir", container_workdir,
+            container_image,
+            "sh", f"{self.script}.sh",
```

Comment:
```suggestion
            "--entrypoint={self.script}.sh",
```

Reply:
this does not work for me.
```
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "--entrypoint=build_sloth.sh": executable file not found in $PATH: unknown.
```

Reply:
`"--entrypoint={mountpoint}/{self.script}.sh"` then

Reply:
this also does not work since the build scripts amaranth generates do not have a shebang

Reply:
Oh. I thought they do. That... seems a problem in its own right, actually.

Reply:
Should be fine after https://github.com/amaranth-lang/amaranth/pull/1012.

Reply:
All right. Let's just drop that part then.